Ignore EPIPE when flushing stdout stderr

lamby opened the pull request:
##### Issue Type:
- Bugfix Pull Request
##### Ansible Version:

2.1.0 (git)
##### Summary:

Ignore EPIPE to avoid tracebacks when piping output to other commands. For example:

```
  $ ansible web --list-hosts | head -n1
```

Such a pipe target will close up shop early when its seen enough input,
causing ansible to print an ugly traceback.
##### Example output:

```
  $ ansible web --list-hosts | head -n1
  hosts (7):
  ERROR! Unexpected Exception: [Errno 32] Broken pipe
  Traceback (most recent call last):
    File "/home/lamby/git/private/lamby-ansible2/.venv/bin/ansible", line 114, in <module>
      display.display("to see the full traceback, use -vvv")
    File "/home/lamby/git/private/lamby-ansible2/.venv/local/lib/python2.7/site-packages/ansible/utils/display.py", line 133, in display
      sys.stdout.flush()
  IOError: [Errno 32] Broken pipe
  $

  $ ansible web --list-hosts | head -n1 # with patch applied
  hosts (7):
  $
```

sivel commented:
This is failing Python 3 tests.  This is due to using `, e` when catching exceptions.

Outside of module code it is preferred to use `as e` instead.

lamby commented:
Thanks @bcoca. Just updated with that - could you review again please?

bcoca commented on lib/ansible/utils/display.py:
```diff
@@ -128,12 +129,15 @@ def display(self, msg, color=None, stderr=False, screen_only=False, log_only=Fal
-            else:
-                sys.stderr.write(msg2)
-                sys.stderr.flush()
+            fileobj = sys.stderr if stderr else sys.stdout
```
no trailing conditionals, people tend to miss these (i'm guilty of having written too many of em)

lamby committed:
Alias fileobj.

lamby committed:
Ignore EPIPE to avoid tracebacks when piping output to other commands

For example:

  $ ansible web --list-hosts | head -n1
  hosts (7):
  ERROR! Unexpected Exception: [Errno 32] Broken pipe
  Traceback (most recent call last):
    File "/home/lamby/git/private/lamby-ansible2/.venv/bin/ansible", line 114, in <module>
      display.display("to see the full traceback, use -vvv")
    File "/home/lamby/git/private/lamby-ansible2/.venv/local/lib/python2.7/site-packages/ansible/utils/display.py", line 133, in display
      sys.stdout.flush()
  IOError: [Errno 32] Broken pipe

Such a pipe target will close up shop early when its seen enough input,
causing ansible to print an ugly traceback.

Signed-off-by: Chris Lamb <chris@chris-lamb.co.uk>

lamby commented:
@bcoca 

> no trailing conditionals

Good point. Pushed - more readable now. I've also expanded the comment.
